src/OpenSage.Game.Tests/Data/Ini/IniFileTests.cs

```diff
@@ -25,38 +26,161 @@ public void CanReadIniFiles()
 
-                    var dataContext = new IniDataContext(entry.FileSystem, gameDefinition.Game);
+                        switch (gameDefinition.Game)
+                        {
+                            case SageGame.Bfme:
+                            case SageGame.Bfme2:
+                            case SageGame.Bfme2Rotwk:
+                                dataContext.LoadIniFile(@"Data\INI\GameData.ini");
+                                break;
+                        }
+                        
 
-                    // BFME I and II need to have GameData.ini loaded before any other INI files,
-                    // because GameData.ini contains global macro definitions.
-                    if (entry.FilePath.ToLowerInvariant() != @"data\ini\gamedata.ini")
-                    {
-                        dataContext.LoadIniFile(@"Data\INI\GameData.ini");
-                    }
+                        foreach (var file in fileSystem.Files)
+                        {
+                            if (Path.GetExtension(file.FilePath).ToLowerInvariant() != ".ini")
+                            {
+                                continue;
+                            }
+
+                            var filename = Path.GetFileName(file.FilePath).ToLowerInvariant();
+
+                            switch (filename)
+                            {
+                                case "buttonsets.ini": // Doesn't seem to be used?
+                                case "scripts.ini": // Only needed by World Builder?
+                                case "commandmapdebug.ini": // Only applies to DEBUG and INTERNAL builds
+                                case "fxparticlesystemcustom.ini": // Don't know if this is used, it uses Emitter property not used elsewhere
+                                case "lightpoints.ini": // Don't know if this is used.
+                                    continue;
+
+                                case "optionregistry.ini": // Don't know if this is used
+                                case "localization.ini": // Don't know if we need this
+                                    switch(gameDefinition.Game)
+                                    {
+                                        case SageGame.Bfme:
+                                        case SageGame.Bfme2:
+                                        case SageGame.Bfme2Rotwk:
+                                            continue;
+                                    }
+                                    break;
+                            }
 
-                    dataContext.LoadIniFile(entry);
-                });
+                            switch (gameDefinition.Game)
+                            {
+                                case SageGame.Bfme2:
+                                    switch (filename)
+                                    {
+                                        case "ingamenotificationbox.ini":
```

Comment:
These are the lists of files still to do?

Reply:
yes, will add a comment there